### PR TITLE
[ZEPPELIN-1483] Zeppelin home page list notebooks doesn't show notebook with group permission

### PR DESCRIPTION
**Pull request by prabhjyotsingh**
### What is this PR for?
Zeppelin home page list notebooks doesn't show notebook with group permission

### What type of PR is it?
[Bug Fix]

### Todos
* [x] - consume userAndRole instead of AuthenticationInfo

### What is the Jira issue?
* [ZEPPELIN-1483](https://issues.apache.org/jira/browse/ZEPPELIN-1483)

### How should this be tested?
In current scenario only those notebook lists that have direct user permission, those with group does not list up, but if user have link to those notebook, it can still be accessed. 
IMO the notebook with group permission should also be listed in the home screen.

### Screenshots (if appropriate)
testgroup

### Questions:
* Does the licenses files need update? n/a
* Is there breaking changes for older versions? n/a
* Does this needs documentation? n/a

### CODE REVIEW

**Comment by prabhjyotsingh** [comment]
Ready for review.
CI fails for #7346.9 with; which is unrelated

```
[31m- should provide onclick method *** FAILED ***[0m
[31m  The code passed to eventually never returned normally. Attempted 1 times over 310.82543 milliseconds. Last failure message: 0 was not equal to 1. (AbstractAngularElemTest.scala:72)[0m
[32mAngularElem[0m
[32m- should print angular display directive only once in a paragraph[0m
[32mAngularElem[0m
[32m- should bind angularObject to ng-model directive[0m
[32mAngularElem[0m
[32m- should able to disassociate AngularObjects[0m
[32mAngularElem[0m
[32m- should allow access to InterpreterContext inside of callback function[0m
[32mAngularElem[0m
[32m- should able to be created from implicit conversion[0m
[32mAngularElemTest:[0m
[32mAngularElem[0m
[32m- should provide onclick method[0m
[32mAngularElem[0m
[32m- should print angular display directive only once in a paragraph[0m
[32mAngularElem[0m
[32m- should bind angularObject to ng-model directive[0m
[32mAngularElem[0m
[32m- should able to disassociate AngularObjects[0m
[32mAngularElem[0m
[32m- should allow access to InterpreterContext inside of callback function[0m
[32mAngularElem[0m
[32m- should able to be created from implicit conversion[0m
[36mRun completed in 1 second, 221 milliseconds.[0m
[36mTotal number of tests run: 18[0m
[36mSuites: completed 5, aborted 0[0m
[36mTests: succeeded 17, failed 1, canceled 0, ignored 0, pending 0[0m
[31m*** 1 TEST FAILED ***[0m
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Zeppelin: Interpreter .............................. SUCCESS [ 12.937 s]
[INFO] Zeppelin: Zengine .................................. SUCCESS [  5.803 s]
[INFO] Zeppelin: Display system apis ...................... FAILURE [  2.906 s]
[INFO] Zeppelin: Spark dependencies ....................... SKIPPED
[INFO] Zeppelin: Spark .................................... SKIPPED
[INFO] Zeppelin: Server ................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 22.748 s
[INFO] Finished at: 2016-09-25T05:00:13+00:00
[INFO] Final Memory: 30M/463M
[INFO] ------------------------------------------------------------------------
```

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java [hunk 1]
```diff
@@ -69,6 +40,15 @@
...
 import org.apache.zeppelin.user.Credentials;
+import org.quartz.*;
+import org.quartz.impl.StdSchedulerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
```
Comment (corneadoug) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java]: Not needed if we include `java.util.*`

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java [hunk 2]
```diff
@@ -69,6 +40,15 @@
 import org.apache.zeppelin.search.SearchService;
 import org.apache.zeppelin.user.AuthenticationInfo;
 import org.apache.zeppelin.user.Credentials;
+import org.quartz.*;
+import org.quartz.impl.StdSchedulerFactory;
```
Comment (corneadoug) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java]: Not needed if we include `org.quartz.*`

File: zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java [hunk 3]
```diff
@@ -17,49 +17,36 @@
...
+import org.apache.zeppelin.types.InterpreterSettingsList;
 import org.apache.zeppelin.user.AuthenticationInfo;
+import org.apache.zeppelin.utils.InterpreterBindingUtils;
 import org.apache.zeppelin.utils.SecurityUtils;
+import org.quartz.CronExpression;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
```
Comment (corneadoug) [inline zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java]: Not needed if we include `javax.ws.rs.*`

**Comment by prabhjyotsingh** [comment]
Thank you @corneadoug, have fixed it.

**Comment by khalidhuseynov** [comment]
adding group permissions to listed notes makes sense. tested and works as expected. maybe would makes sense to add integration tests to differentiate the cases of user and groups, maybe in diffent PR as improvement. CI looks irrelevant, LGTM

**Comment by prabhjyotsingh** [comment]
Merging this if no more discussion.

**Pull request closed without merging**